src/sentry/scim/endpoints/members.py

```diff
@@ -372,14 +375,31 @@ def post(self, request: Request, organization) -> Response:
         - The API also does not support setting secondary emails.
         """
 
+        if (
+            features.has("organizations:scim-orgmember-roles", organization, actor=None)
+            and "sentryOrgRole" in request.data
+            and request.data["sentryOrgRole"]
+        ):
+            role = request.data["sentryOrgRole"].lower()
+        else:
+            role = organization.default_role
+
+        # Allow any role as long as it doesn't have `org:admin` permissions
+        allowed_roles = {role for role in roles.get_all() if not role.has_scope("org:admin")}
```

Comment:
Nice. Even though this still may be a little dicey in the long run, it's good that we're tying it to a scope and not hard-coding any assumptions about the structure of `SENTRY_ROLES`.